third_party/CMakeLists.txt

```diff
@@ -1,6 +1,11 @@
+
+# Suppress warnings in third party code.
```

Comment:
I think this should still be moved back to the top level file and managed by adding/removing flags to CXX_FLAGS

Reply:
@sstamenova @gongsu832 I have now moved the logic in the top level `cmake` file and removed the `cmake` file in the `third_party` directory. Last night the build on Win was fine, no warning from third party code. I have just rebased the branch and that triggered another build on all CI bots. 

Could you please take another look at the code to see if you are fine with it in this current form. ?